graphbolt/src/utils.h

```diff
@@ -19,6 +19,13 @@ inline bool is_accessible_from_gpu(torch::Tensor tensor) {
   return tensor.is_pinned() || tensor.device().type() == c10::DeviceType::CUDA;
 }
 
+/**
+ * @brief Checks whether the tensor is stored on the GPU.
+ */
+inline bool is_on_gpu(torch::Tensor tensor) {
```

Comment:
this raises a question: how to use GPU for sampling automatically? we need to make sure `graph` is pinned or on gpu and `nodes` should be on gpu?

Reply:
this is related to what will the `gb.DataLoader` interface be like, namely, how to choose sampling on GPU or CPU by user.

Reply:
If graph is pinned or is on the device and if nodes is on device, we use gpu sampling.

Reply:
It is already automatic, see the cuda examples PR.

Reply:
To dispatch to CUDA, at least one input tensor needs to be on GPU, and the rest either pinned or on GPU.

Reply:
I don't think it's automatic as user needs to call `graph.pin_memory_()` and add `copy_to()` datapipe to copy `nodes` to GPU manually in the examples you showed. This requires extra coding.

Reply:
To use the GPU, we need to insert a CopyTo at the end of the data pipes anyway. What is the difference?

Reply:
Do you think we can do better?

Reply:
we could add such copy_to datapipe within `gb.DataLoader`(including pinning graph) and user just need to specify a target device argument for dataloader.

Reply:
We can discuss this in an issue or on Slack.